src/Markdig/Helpers/StringSlice.cs

```diff
@@ -312,16 +304,15 @@ public bool SkipSpacesToEndOfLineOrEndOfDocument()
         /// </returns>
         public bool TrimStart()
         {
-            // Strip leading spaces
-            for (; Start <= End; Start++)
-            {
-                if (Start < Text.Length
-                    && !Text[Start].IsWhitespace())
-                {
-                    break;
-                }
-            }
-            return IsEmpty;
+            string text = Text;
+            int end = End;
+            int i = Start;
+
+            while (i <= end && (uint)i < (uint)text.Length && text[i].IsWhitespace())
```

Comment:
I might be missing something, but why not leverage the `ReadOnlySpan<T>.Trim` extension (from `MemoryExtensions`) here instead, which is also automatically vectorized? Same for the other trim methods from this type.

Reply:
I would expect the 99% case here to be 0 - 2 spaces, so creating the span would be more overhead.

I have played around with other unsafe helpers (that could be used in `Match` for example), that would take `ref char` to avoid the cost of a span creation, but that's for a different issue/pr.